File: java/src/main/java/com/android/frame/mvc/AATest/TestActivityMvc.java
package com.android.frame.mvc.AATest;

import android.os.Bundle;

import com.android.frame.mvc.BaseActivity;
import com.android.java.databinding.ActivityCommonLayoutBinding;
import com.android.util.CommonLayoutUtil;

/**
 * Created by xuzhb on 2021/1/1
 * Desc:
 */
public class TestActivityMvc extends BaseActivity<ActivityCommonLayoutBinding> {

    @Override
    public void handleView(Bundle savedInstanceState) {
        CommonLayoutUtil.initCommonLayout(this, "MVC框架", "BaseActivity", "BaseListActivity", "BaseFragment", "BaseListFragment");
    }

    @Override
    public void initListener() {
        binding.btn1.setOnClickListener(v -> {
            startActivity(TestMvcActivity.class);
        });
        binding.btn2.setOnClickListener(v -> {
            startActivity(TestMvcListActivity.class);
        });
        binding.btn3.setOnClickListener(v -> {
            Bundle bundle = new Bundle();
            bundle.putInt("fragment_type", 1);
            startActivity(TestFragmentMvc.class, bundle);
        });
        binding.btn4.setOnClickListener(v -> {
            Bundle bundle = new Bundle();
            bundle.putInt("fragment_type", 2);
            startActivity(TestFragmentMvc.class, bundle);
        });
    }

    @Override
    public ActivityCommonLayoutBinding getViewBinding() {
        return ActivityCommonLayoutBinding.inflate(getLayoutInflater());
    }

}

File: kotlin/src/main/java/com/android/frame/mvc/AATest/TestActivityMvc.kt
package com.android.frame.mvc.AATest

import android.os.Bundle
import com.android.basicproject.databinding.ActivityCommonLayoutBinding
import com.android.frame.mvc.BaseActivity
import com.android.util.initCommonLayout

/**
 * Created by xuzhb on 2020/12/30
 * Desc:
 */
class TestActivityMvc : BaseActivity<ActivityCommonLayoutBinding>() {

    override fun handleView(savedInstanceState: Bundle?) {
        initCommonLayout(this, "MVC框架", "BaseActivity", "BaseListActivity", "BaseFragment", "BaseListFragment")
    }

    override fun initListener() {
        binding.btn1.setOnClickListener {
            startActivity(TestMvcActivity::class.java)
        }
        binding.btn2.setOnClickListener {
            startActivity(TestMvcListActivity::class.java)
        }
        binding.btn3.setOnClickListener {
            startActivity(TestFragmentMvc::class.java, Bundle().apply { putInt("fragment_type", 1) })
        }
        binding.btn4.setOnClickListener {
            startActivity(TestFragmentMvc::class.java, Bundle().apply { putInt("fragment_type", 2) })
        }
    }

    override fun getViewBinding() = ActivityCommonLayoutBinding.inflate(layoutInflater)

}